db/music_pg.sql

```diff
@@ -0,0 +1,373 @@
+CREATE TABLE "submission" (
+  "submission_id" uuid PRIMARY KEY DEFAULT (gen_random_uuid()),
+  "participant_id" int NOT NULL,
+  "bracket_id" int NOT NULL,
+  "song_id" int NOT NULL,
+  "order" int NOT NULL,
+  "date_submitted" timestamp NOT NULL DEFAULT (now()),
+  "active" bool_int NOT NULL,
+  "is_dupe" bool_int NOT NULL,
+  "title" text NOT NULL,
+  "version" text,
+  "artist" text,
+  "collaborators" text,
+  "composer" text,
+  "link" text NOT NULL,
+  "genre" text,
+  "emoji" text,
+  "comments" text
+);
+
+CREATE TABLE "submission_history" (
+  "submission_history_id" uuid PRIMARY KEY DEFAULT (gen_random_uuid()),
+  "date_update" timestamp NOT NULL,
+  "note_update" text NOT NULL,
+  "submission_id" int NOT NULL,
+  "participant_id" int NOT NULL,
+  "bracket_id" int NOT NULL,
+  "song_id" int NOT NULL,
+  "order" int NOT NULL,
+  "date_submitted" timestamp NOT NULL DEFAULT (now()),
+  "active" bool_int NOT NULL,
+  "is_dupe" bool_int NOT NULL,
+  "title" text NOT NULL,
+  "version" text,
+  "artist" text,
+  "collaborators" text,
+  "composer" text,
+  "link" text NOT NULL,
+  "genre" text,
+  "emoji" text,
+  "comments" text
+);
+
+CREATE TABLE "link_relation" (
+  "link_realtion_id" uuid PRIMARY KEY DEFAULT (gen_random_uuid()),
+  "relation_id" int NOT NULL,
+  "link_id" int NOT NULL,
+  "type_link" text NOT NULL
+);
+
+CREATE TABLE "link" (
+  "link_id" uuid PRIMARY KEY DEFAULT (gen_random_uuid()),
+  "url" text NOT NULL,
+  "region" text[]
+);
+
+CREATE TABLE "song" (
+  "song_id" uuid PRIMARY KEY DEFAULT (gen_random_uuid()),
+  "title" text NOT NULL,
+  "title_english" text,
+  "version" text,
+  "artist" text NOT NULL,
+  "collaborators" text,
+  "composer" text,
+  "year_min" int,
+  "year_max" int
+);
+
+CREATE TABLE "song_relation" (
+  "song_relation_id" INT GENERATED BY DEFAULT AS IDENTITY PRIMARY KEY,
+  "song_from_id" int NOT NULL,
```

Comment:
Very minor personal preference to consider, but I like the `parent` and `child` qualifiers here, and again would like to swap the order around to make it `parent_song_id` and `child_song_id`, putting the most important/distinguishing thing at the front here so that you notice it slightly easier. The second bit may, in fact, be somewhat at odds with the convention I put forth above for `update_note` and `update_date`, though I think that given the nature of that table and the nature of this table, the difference makes sense to me.

Reply:
I guess an addendum here about `to`/`from` vs `child`/`parent` is that `song_to_id` and `song_from_id` sound like function names where you're given a song and convert it to an ID or vice versa, where as `child` and `parent` are more obvious as qualifiers for the specific type of song than actions performed using them.

Reply:
Yeah, I wasn't please with my approach. I don't entirely love parent/child, either, for some reason, but it'll probably work.